sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -38,7 +38,7 @@ import org.apache.spark.sql.execution.datasources.jdbc._
 import org.apache.spark.sql.execution.datasources.json.TextInputJsonDataSource
 import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
 import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Utils
-import org.apache.spark.sql.sources.v2.{BatchReadSupportProvider, DataSourceOptions, DataSourceV2}
+import org.apache.spark.sql.sources.v2._
```

Comment:
Nit: using wildcard imports makes it harder to review without an IDE because it is more difficult to find out where symbols come from.

Reply:
I do think this one is too nitpicking. If this gets long it should be wildcard. Use an IDE for large reviews like this if needed.

Reply:
It's the IDE that turns it into wildcard, because it gets too long.

Reply:
I am using an IDE for this review, but this makes future reviews harder. I realize it isn't a major issue, but I think it is a best practice to not use wildcard imports.